src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -0,0 +1,289 @@
+                const auto actual_before = parent_output.get_element_type();
+                if (actual_before != required_after) {
+                    input_precisions_were_changed = true;
+                    auto existing_convert = ngraph::as_type<ngraph::snippets::op::ConvertSaturation>(
+                        parent_output.get_node());
+                    if (existing_convert == nullptr) {
+                        auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                            parent_output,
+                            required_after);
+                        ngraph::copy_runtime_info(parent_output.get_node_shared_ptr(), convert);
+                        op->set_argument(op_input.get_index(), convert);
+                    } else {
+                        const auto actual_before = existing_convert->get_input_source_output(0).get_element_type();
+                        const auto actual_after = existing_convert->output(0).get_element_type();
+                        if (can_be_removed(actual_before, actual_after, required_after)) {
+                            existing_convert->output(0).replace(parent_output);
+                        } else {
+                            if (can_be_fused(actual_after, required_after)) {
+                                auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                                    existing_convert->get_input_node_shared_ptr(0),
+                                    required_after);
+                                ngraph::copy_runtime_info(parent_output.get_node_shared_ptr(), convert);
+                                op->set_argument(op_input.get_index(), convert);
+                            } else {
+                                auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                                    existing_convert->output(0),
+                                    required_after);
+                                ngraph::copy_runtime_info(existing_convert->shared_from_this(), convert);
+                                op->set_argument(op_input.get_index(), convert);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (input_precisions_were_changed || (type_relaxed_node != nullptr)) {
+            // update output precision
+            std::vector<element::Type> op_output_types;
+            for (auto& output : op->outputs()) {
+                op_output_types.push_back(output.get_element_type());
+            }
+
+            if (type_relaxed_node != nullptr) {
+                // to keep previous functionality
+                // unary and binary element-wise operations are supported
+                // will be replaced to snippets opset later
+                const auto op_element_type = op->get_input_source_output(0).get_element_type();
```

Comment:
Add please TODO that we should add new Snippets operations for `TypeRelaxed` nodes. And create the ticket please if there isn't. Thanks!

Reply:
done, thanks

Reply:
updated, thanks